core/runtime/src/main/java/io/quarkiverse/langchain4j/RegisterAiService.java

```diff
@@ -0,0 +1,156 @@
+package io.quarkiverse.langchain4j;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.util.function.Supplier;
+
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.memory.chat.ChatMemoryProvider;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.moderation.ModerationModel;
+import dev.langchain4j.retriever.Retriever;
+import dev.langchain4j.service.AiServices;
+
+/**
+ * Used to create Langchain4j's {@link AiServices} in a declarative manner that the application can then use simply by
+ * using the class as a bean.
+ * Under the hood Langchain4j's {@link AiServices#builder(Class)} is called
+ * while also providing the builder with the proper {@link ChatLanguageModel}, {@code tools} beans,
+ * {@link ChatMemory} or {@link ChatMemoryProvider}, {@link ModerationModel} and {@link Retriever}
+ */
+@Retention(RUNTIME)
+@Target(ElementType.TYPE)
+public @interface RegisterAiService {
+
+    /**
+     * Configures the way to obtain the {@link ChatLanguageModel} to use.
+     * If not configured, the CDI bean implementing the model is looked up.
+     * Such a bean provided automatically by extensions such {@code quarkus-langchain4j-openai} and
+     * {@code quarkus-langchain4j-hugging-face}
```

Comment:
We might need to add a bit of validation: 
- if multiple models are there and this value is not set -> BOOM - this can be done at build time.

Reply:
This already happens IIRC